src/lib/core/theming/test-css-variables-theme.scss

```diff
@@ -0,0 +1,24 @@
+@import './all-theme';
+
+// Recursively replaces all of the values inside a Sass map with a different value.
+@function replace-all-values($palette, $replacement) {
+  $output: ();
+
+  @each $key, $value in $palette {
+    @if (type-of($value) == 'map') {
+      $output: map-merge(($key: replace-all-values($value, $replacement)), $output);
+    }
+    @else {
+      $output: map-merge(($key: $replacement), $output);
+    }
+  }
+
+  @return $output;
+}
+
+// Theme used to test that our themes would compile if the colors were specified as CSS variables.
+._demo-css-variables-theme {
+  $palette: mat-palette($mat-blue-grey);
+  $theme: mat-dark-theme($palette, $palette, $palette);
+  @include angular-material-theme(replace-all-values($theme, var(--test-var)));
```

Comment:
I'm curious how things would actually look with the changes you've made, could we make a similar function that stores off the current values in the map (e.g. `--test-var-#{n}: $value`) and then updates the map with variables referring to the original value (e.g. `var(--test-var-#{n})`). That way we could compare how the app looks with the original theme and with the variable alias theme

Reply:
We should be able to get it to work, but I'd rather get this in first so we don't continue introducing new failures.

Reply:
I suppose it doesn't make things any worse for users